docs/source/src/rust/user-guide/expressions/operations.rs

```diff
@@ -17,8 +17,8 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         .clone()
         .lazy()
         .select([
-            (col("nrs") + lit(5)).alias("nrs + 5"),
-            (col("nrs") - lit(5)).alias("nrs - 5"),
+            (col("nrs") + lit(5).cast(DataType::Int32)).alias("nrs + 5"),
```

Comment:
I think this should be `typed_lit`, but what is the failure here?

Reply:
without the cast, the error is:

`SchemaMismatch: failed to determine supertype of i32 and dyn int`

I may be wrong, but it doesn't look like `typed_lit` is exposed.

Reply:
actually, the cast is not needed. This example required the `dtype-i8` feature as `materialize_smallest_dyn_int` ends up converting to `i8`.

Reply:
@ritchie46 this is an issue I've encountered and luke's conclusion is correct/what you folks told me: that I needed the feature `dtype-i8` but that something needed fixing because it wasn't supposed to require a feature.